src/stores/notebook.ts

```diff
@@ -105,6 +114,39 @@ export class Notebook {
     return this.state.kernelId;
   }
 
+  async loadNotebook() {
+    try {
+      const notebook = await invoke<NotebookType>("get_notebook", {
+        path: this.path,
+      });
+
+      this.state.cellIds = notebook.cells.map((cell) => cell.id);
```

Comment:
Mutations to state need to go through actions in `NotebookStoreActions` which use immer. Can't mutate state directly here since it isn't compatible with Zustand's reactivity

(it doesn't know which components to rerender when you make the mutation directly instead of using an action)

Reply:
Will update! Haven’t used immer / zustand so will look into this, thanks for flagging

Reply:
Thanks :D other than that it looks good, I think. Having the tauri apps plugins in package.json will be nice if we need them in the future.

I can take a pass at some UI TODOs later for the loading states. Currently stuck in language server hell :D